src/logger/Logger.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+import os
+from enum import Enum
+
+class Severity(Enum):
+    DEBUG = 1
+    INFO = 2
+    WARNING = 3
+    ERROR = 4
+
+class Logger:
+    
+    def __init__(self, debug=False, filename='log.txt'):
+        self.debug = debug
+        self.filename = filename
+        self._f = open(self.filename,'a+')
+
+        self._f.write('[{}]  {}  {} \n'.format( 'INFO', str(datetime.datetime.now()), 'Logger started.'))
+
+
+    def log(self, message, severity=Severity.INFO):
+        timestamp = datetime.datetime.now()
+
+        if severity == Severity.ERROR:
+            self._f.write('[{}]  {}  {} \n'.format('ERROR', str(timestamp), message))
+        elif severity == Severity.INFO:
+            self._f.write('[{}]  {}  {} \n'.format('INFO', str(timestamp), message))
+        elif severity == Severity.WARNING:
+            self._f.write('[{}]  {}  {} \n'.format('WARNING', str(timestamp), message))
+        elif severity == Severity.DEBUG:
+            if self.debug:
+                self._f.write('[{}]  {}  {} \n'.format('DEBUG', str(timestamp), message))
+                
+        self._f.flush()
+        os.fsync(self._f)
```

Comment:
So we don't close the connection to the file, which I like. But I thought we wanted to not write on every log as that involves IO. Do I have this wrong, or do I misunderstand this?

Reply:
I thought we decided that it was more important to make sure that log entries actually end up in the log file in the event of a server crash. In any case changing that requirement would require more work that I think I can do before the end of sprint so I'm just going to merge for now.